dashboard/src/actions/quisbyChartActions.js

```diff
@@ -0,0 +1,124 @@
+import * as TYPES from "./types.js";
+
+import { DANGER, ERROR_MSG } from "assets/constants/toastConstants";
+
+import API from "../utils/axiosInstance";
+import { showToast } from "./toastActions";
+import { uriTemplate } from "../utils/helper";
+
+const chartTitleMap = {
+  gb_sec: "Bandwidth",
+  trans_sec: "Transactions/second",
+  usec: "Latency",
+};
+
+export const getQuisbyData = (dataset) => async (dispatch, getState) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+    const endpoints = getState().apiEndpoint.endpoints;
+    dispatch({
+      type: TYPES.SET_ACTIVE_RESOURCEID,
+      payload: dataset.resource_id,
+    });
+    const response = await API.get(
+      uriTemplate(endpoints, "datasets_visualize", {
+        dataset: dataset.resource_id,
+      })
+    );
+    if (response.status === 200 && response.data.json_data) {
+      dispatch({
+        type: TYPES.GET_QUISBY_DATA,
+        payload: response.data.json_data,
+      });
+      dispatch({
+        type: TYPES.IS_UNSUPPORTED_TYPE,
+        payload: "",
+      });
+      dispatch(parseChartData());
+    }
+    dispatch({ type: TYPES.COMPLETED });
+  } catch (error) {
```

Comment:
If `response.status` is not `200` or if `response.data.json_data` is false-y, we don't set `IS_UNSUPPORTED_TYPE`, and we silently "complete"?  Shouldn't there be an `else` for the `if` at line 28 which displays some sort of error?  Or, perhaps, it should `throw` an error which we'll catch at line 40 and trigger the existing error handling?

Reply:
So _error_ responses (such as "Unsupported type"/`400`, and, I hope, all `5xx` responses) will be properly handled -- that's good -- and, hopefully, `1xx` and `3xx` responses will be retried or will otherwise never get here. 
 That narrows the field down to just the `2xx` responses...which leads me to wonder why you have an `if` statement here.

That is, there are two approaches:  one is to trust that Quisby is well-behaved, that it will never return any `2xx` response other than `200`, and that, when it returns `200` it will always return a properly-populated response payload including a non-empty `json_data` field; the other approach is to be paranoid and be prepared to handle _any_ response, including an apparently successful one which is lacking anything in the `json_data` field.  I would have chosen the latter path, but you seem to have split the difference!  That is, if you're going to trust that Quisby is well-behaved, then there is no need for the `if` statement (and no need for the `else` that I want...); it's only if you don't trust Quisby that you need the `if` statement...but, then, you need to do something if the condition fails!  🙃